File: TP1 - C++/data/models/orthodontist.cpp
#include "../../domain/entities/orthodontist.hpp"

Orthodontist::Orthodontist(string id):
   Employee(id){};

Orthodontist::~Orthodontist(){};

string Orthodontist::getId(){
    return this->id;
} 

Schedule Orthodontist::getSchedule(){
    return this->schedule;
} 

Assistent Orthodontist::getAssistent(){
    return this->assistent;
}

void Orthodontist::setSchedule(Schedule schedule){
    this->schedule = schedule;
}

void Orthodontist::setAssistent(Assistent assistent){
    this->assistent = assistent;
}

void Orthodontist::printSchedule(){
    this->schedule.print();
}

void Orthodontist::addConsultation(Consultation consultation){
    this->schedule.addConsultation(consultation);
}

File: TP2 - Java/src/domain/entities/Orthodontist.java
package domain.entities;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.UUID;

public class Orthodontist extends Employee {
  private Schedule schedule;
  private Assistent assistent;

  public Orthodontist(String id, String name, String login, String password, UserPermissions userPermissions) {
    super(id, name, login, password, userPermissions);
    this.schedule = new Schedule(UUID.randomUUID().toString(), new ArrayList<Consultation>());
  }

  public Schedule getSchedule() {
    return schedule;
  }

  public Assistent getAssistent() {
    return assistent;
  }

  public void setSchedule(Schedule schedule) {
    this.schedule = schedule;
  }

  public void setAssistent(Assistent assistent) {
    this.assistent = assistent;
  }
}
